java/refactoring.java/src/org/netbeans/modules/refactoring/java/WhereUsedBinaryElement.java

```diff
@@ -64,14 +64,12 @@ public String getDisplayText() {
     public Lookup getLookup() {
         Icon icon = null;
         try {
-            ImageIcon imageIcon = new ImageIcon(DataObject.find(fo).getNodeDelegate().getIcon(BeanInfo.ICON_COLOR_16x16));
+            icon = ImageUtilities.image2Icon(DataObject.find(fo).getNodeDelegate().getIcon(BeanInfo.ICON_COLOR_16x16));
             Boolean inTestFile = ElementGripFactory.getDefault().inTestFile(fo);
             if (Boolean.TRUE == inTestFile) {
-                Image mergeImages = ImageUtilities.mergeImages(imageIcon.getImage(),
-                        ImageUtilities.loadImageIcon("org/netbeans/modules/refactoring/java/resources/found_item_test.png", false).getImage(), 4, 4);
-                imageIcon = new ImageIcon(mergeImages);
+                icon = ImageUtilities.image2Icon(ImageUtilities.mergeImages(ImageUtilities.icon2Image(icon),
+                        ImageUtilities.loadImage("org/netbeans/modules/refactoring/java/resources/found_item_test.png", false), 4, 4));
             }
-            icon = imageIcon;
         } catch (DataObjectNotFoundException ex) {
```

Comment:
similar here. I think `ImageIcon icon` would allow to skip the call to `icon2image`.

although the lookup registration later would have to be changed to `icon.getImage()`

Reply:
I can simplify it a bit here; keeping the ImageIcon from loadImageIcon while keeping image2Icon instead of the "new ImageIcon" constructor.